sql/chunk.sql

```diff
@@ -27,23 +27,57 @@ $BODY$
 DECLARE
     partition_epoch_row _timescaledb_catalog.partition_epoch;
     chunk_row           _timescaledb_catalog.chunk;
-    time_interval BIGINT;
+    time_interval       BIGINT;
+    start_distinct_ctn  BIGINT;
+    end_distinct_ctn    BIGINT;
 BEGIN
+
     SELECT pe.*
     INTO partition_epoch_row
     FROM _timescaledb_catalog.partition p
     INNER JOIN _timescaledb_catalog.partition_epoch pe ON (p.epoch_id = pe.id)
     WHERE p.id = partition_id
     FOR SHARE;
 
+    -- Check if there are chunks in other partitions.
+    -- If so, use their time range to keep chunks time aligned
+
+    SELECT COUNT(DISTINCT(start_time)), COUNT(DISTINCT(end_time))
```

Comment:
You could do 
```SELECT DISTINCT(start_time), DISTINCT(end_time) INTO STRICT table_start, table_end ``` 
That would throw an error if not 1 row returned (and you could catch and ignore the 0 row case). 

In any case, what you have now works so I'm approving. No need to change if you don't want to...